### PR TITLE
DA: DispersedBlobInfo trait in nomos-core and kzgrs-backend

### PR DESCRIPTION
**Pull request by bacv**
Modify DaEncoder and DaDispersal traits to work with DispersedBlobInfo. Before moving to the cli app, changes needs to be applied to the core and services.

### CODE REVIEW

File: nomos-core/src/da/mod.rs [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
+    fn verify(&self, blob: &Self::DaBlob) -> Result<(), Self::Error>;
 }
 
 pub trait DaDispersal {
     type EncodedData;
-    type Certificate;
+    type Error;
 
-    fn disperse(&self, encoded_data: Self::EncodedData) -> Result<Self::Certificate, impl Error>;
+    fn disperse(&self, encoded_data: Self::EncodedData) -> Result<(), Self::Error>;
```
Comment (danielSanchezQ) [inline nomos-core/src/da/mod.rs:24]: We could have left the certificate that way it would be up to implementation to use `()`. But his way is closer to the actual spec.

File: nomos-core/src/da/blob/info.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use super::metadata::Metadata;
+
+pub trait DispersedBlobInfo: Metadata {
+    type BlobId;
+
+    fn blob_id(&self) -> Self::BlobId;
+    fn size(&self) -> usize;
```
Comment (danielSanchezQ) [inline nomos-core/src/da/blob/info.rs:7]: I'm wondering now if we could have just go for the blob struct directly. But we will see it when composing the final implementation.

File: nomos-da/kzgrs-backend/benches/verifier.rs [hunk 3]
```diff
@@ -26,7 +26,7 @@ fn verify<const SIZE: usize>(bencher: Bencher, column_size: usize) {
 
             let encoder = DaEncoder::new(params);
             let data = rand_data(SIZE * MB / DaEncoderParams::MAX_BLS12_381_ENCODING_CHUNK_SIZE);
-            let encoded_data = encoder.encode(&data).unwrap();
+            let encoded_data = nomos_core::da::DaEncoder::encode(&encoder, &data).unwrap();
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/benches/verifier.rs:29]: Why this change?

**Review by danielSanchezQ — APPROVED** [review]
Necessary refactor, thanks for the effort.

File: nomos-core/src/da/blob/info.rs (hunk 2 above)
Comment (bacv) [reply nomos-core/src/da/blob/info.rs:7]: If we use blob directly, the metadata will need to be optional, I think similar case was with the Certificate.

File: nomos-da/kzgrs-backend/benches/verifier.rs (hunk 3 above)
Comment (bacv) [reply nomos-da/kzgrs-backend/benches/verifier.rs]: I think I had a rust-analyzer bit laggy, because it didn't let recognize the `encode` method from the DaEncoder even after the import. Tried now and it works, fixed.
